src/finder/gitHubFinder.ts

```diff
@@ -28,44 +29,81 @@ export class GitHubCodeBlockFinder implements CodeFinder {
 
 export class GitHubFileViewFinder implements CodeFinder {
   private readonly URL_REGEX = /^https:\/\/github\.com\/.*\/.*\.(plantuml|pu|puml|wsd)(\?.*)?$/;
+  private readonly EXCLUDE_URL_REGEX = /^https:\/\/github\.com\/.*\/edit\/.*/;
   private readonly INCLUDE_REGEX = /^\s*!include\s+(.*\.(plantuml|pu|puml|wsd))\s*$/;
-  private readonly EDIT_URL_REGEX = /^https:\/\/github\.com\/.*\/edit\/.*/;
+  private readonly INCLUDESUB_REGEX = /^\s*!includesub\s+(.*\.(plantuml|pu|puml|wsd))!(.*)\s*$/;
 
   canFind(webPageUrl: string): boolean {
-    if (this.EDIT_URL_REGEX.test(webPageUrl)) return false;
+    if (this.EXCLUDE_URL_REGEX.test(webPageUrl)) return false;
     return this.URL_REGEX.test(webPageUrl);
   }
 
   async find(webPageUrl: string, $root: JQuery<Node>): Promise<UmlCodeContent[]> {
     const $texts = $root.find(`div[itemprop='text']:not([${Constants.ignoreAttribute}])`);
-    const dirUrl = webPageUrl.replace(/\/[^/]*\.(plantuml|pu|puml|wsd)(\?.*)?$/, '');
     const result = [];
     for (let i = 0; i < $texts.length; i++) {
       const $text = $texts.eq(i);
-      let fileText = '';
       const $fileLines = $text.find('tr');
-      for (let i = 0; i < $fileLines.length; i++) {
-        const lineText = $fileLines.eq(i).find("[id^='LC'").text();
-        const match = this.INCLUDE_REGEX.exec(lineText);
-        if (match !== null) {
-          const includedFileText = await this.getIncludedFileText(`${dirUrl}/${match[1]}`);
-          fileText += includedFileText || '';
-        } else {
-          fileText += lineText + '\n';
-        }
-      }
+      let fileText = [...Array($fileLines.length).keys()]
+        .map((lineno) => $fileLines.eq(lineno).find("[id^='LC'").text() + '\n')
+        .join('');
+      fileText = await this.preprocessIncludeDirective(webPageUrl, fileText);
+      fileText = await this.preprocessIncludesubDirective(webPageUrl, fileText);
       result.push({ $text, text: fileText });
     }
     return result;
   }
 
-  private async getIncludedFileText(fileUrl: string): Promise<string | null> {
-    const response = await fetch(fileUrl);
-    if (!response.ok) return null;
-    const htmlString = await response.text();
-    const $body = $(new DOMParser().parseFromString(htmlString, 'text/html')).find('body');
-    const contents = await this.find(fileUrl, $body);
-    return contents.map((content) => content.text.replace(/@startuml/g, '').replace(/@enduml/g, '')).join('\n');
+  private async preprocessIncludeDirective(webPageUrl: string, fileText: string): Promise<string> {
+    const fileTextLines = fileText.split('\n');
+    const dirUrl = webPageUrl.replace(/\/[^/]*\.(plantuml|pu|puml|wsd)(\?.*)?$/, '');
+
+    const preprocessedLines = [];
+    for (const line of fileTextLines) {
+      const match = this.INCLUDE_REGEX.exec(line);
+      if (!match) {
+        preprocessedLines.push(line);
+        continue;
+      }
+      const includedText = await (async () => {
```

Comment:
I feel this block is a bit complicated (though I can understand `return` is useful).
How about the following code.

```
        const includedFileUrl = `${dirUrl}/${match[1]}`;
        const response = await fetch(includedFileUrl);
        let includedText = '';
        if (response.ok) {
          const htmlString = await response.text();
          const $body = $(new DOMParser().parseFromString(htmlString, 'text/html')).find('body');
          const fileTexts = await this.find(includedFileUrl, $body);
          includedText = fileTexts.map((fileText) => fileText.text.replace(/@startuml/g, '').replace(/@enduml/g, '')).join('\n');
        }
```

Reply:
Or can we rewrite this block as follows?

```
        const includedFileUrl = `${dirUrl}/${match[1]}`;
        const response = await fetch(includedFileUrl);
        if (response.ok) {
          const htmlString = await response.text();
          const $body = $(new DOMParser().parseFromString(htmlString, 'text/html')).find('body');
          const fileTexts = await this.find(includedFileUrl, $body);
          const includedText = fileTexts.map((fileText) => fileText.text.replace(/@startuml/g, '').replace(/@enduml/g, '')).join('\n');
          preprocessedLines.push(includedText);
        }
```